test/unit/data.js

```diff
@@ -625,7 +625,6 @@ QUnit.test( ".data always sets data with the camelCased key (gh-2257)", function
 		div.data( key, val );
 		var allData = div.data();
 		assert.equal( allData[ key ], undefined, ".data does not store with hyphenated keys" );
-		assert.equal( allData[ jQuery.camelCase( key ) ], val, ".data stores the camelCased key" );
```

Comment:
This assertion is very important and shouldn't be removed. Instead, I'd expect the `datas` object to be modified to contain the expected camelCased value, sth like:
```js
datas = {
	"non-empty": {camelCasedKey: "nonEmpty", value: "a string"},
	...
};
```

The same applies to other places where you removed the assertions using `jQuery.camelCase`. It's not the assertions that need removal but `jQuery.camelCase` usage.